tensorflow/contrib/layers/python/layers/layers.py

```diff
@@ -1201,16 +1202,29 @@ def flatten(inputs,
     ValueError: if inputs.dense_shape is wrong.
   """
   with ops.name_scope(scope, 'Flatten', [inputs]) as sc:
-    inputs = ops.convert_to_tensor(inputs)
-    inputs_shape = inputs.get_shape()
-    inputs_rank = inputs_shape.ndims
-    if (inputs_rank is None) or (inputs_rank < 2):
```

Comment:
I would probably keep the check that the inputs have at least 2 dimensions, fail sooner.